### PR TITLE
feat(button): add enhancers

### PR DESCRIPTION
**Pull request by colbr**
## 1. Pull request checklist

- [x] PR name follows [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

## 2. What is the old behaviour? (if applicable)

There was no way to add a start or end enhancer to `Button`.

## 3. What is the new behaviour?

Can now add a start or end enhancer to `Button` through the `startEnhancer` and `endEnhancer` props.

## 4. How can this behaviour be tested? (if applicable)

- Check the `Button` enhancer stories in Storybook
- Run tests for `Button.tsx`

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -52,7 +57,11 @@ export const Button: FC<ButtonProps> = ({
       ...buttonProps
     },
     <>
-      <div className="zui-button-content">{isLoading ? <Spinner className="zui-button-spinner" /> : children}</div>
+      <div className="zui-button-content">
+        {startEnhancer}
+        {isLoading ? <Spinner className="zui-button-spinner" /> : children}
```
Comment (dalefukami) [inline src/components/Button/Button.tsx:62]: Should the loading state still show the icon? I haven't seen the design but that feels weird to me.

**Review by dalefukami — APPROVED** [review]
Just as a general question, do we even need this. Why shouldn't the consumer be in control of what's rendered inside the button by just passing in whatever nodes they want as children?

**Comment by colbr** [comment]
> Just as a general question, do we even need this. Why shouldn't the consumer be in control of what's rendered inside the button by just passing in whatever nodes they want as children?

Oops sorry missed this question. It's standard to pass enhancers in as props - it makes for a simpler API, and encourages correct use of the design system (e.g. the design system specifies 1 enhancer maximum on each side, which would be much less obvious if we dropped `startEnhancer`/`endEnhancer` and left it up to the consumer)

**Comment by colbr** [comment]
Merging now as I have Dale's review. We should change branch protection rules ☺️

**Pull request merged**